script/tool/lib/src/update_snippets_command.dart

```diff
@@ -0,0 +1,217 @@
+          snippetDirectory.deleteSync(recursive: true);
+        }
+      }
+    }
+
+    if (getBoolArg(_failOnChangeFlag)) {
+      final String? stateError = await _validateRepositoryState();
+      if (stateError != null) {
+        printError('README.md is out of sync with its source snippets.\n\n'
+            'If you edited code in README.md directly, you should instead edit '
+            'the example source files. If you edited source files, run the '
+            'repository tooling\'s "$name" command on this package, and update '
+            'your PR with the resulting changes.');
+        return PackageResult.fail(<String>[stateError]);
+      }
+    }
+
+    return PackageResult.success();
+  }
+
+  /// Runs the extraction step to create the snippet files for the given
+  /// example, returning true on success.
+  Future<bool> _extractSnippets(RepositoryPackage example) async {
+    final int exitCode = await processRunner.runAndStream(
+        'dart',
+        <String>[
+          'run',
+          'build_runner',
+          'build',
+          '--config',
+          _buildRunnerConfigName,
+          '--output',
+          _snippetOutputDir,
+          '--delete-conflicting-outputs',
+        ],
+        workingDir: example.directory);
+    return exitCode == 0;
+  }
+
+  /// Runs the injection step to update [targetPackage]'s README with the latest
+  /// snippets from [example], returning true on success.
+  Future<bool> _injectSnippets(
+    RepositoryPackage example, {
+    required RepositoryPackage targetPackage,
+  }) async {
+    final String relativeReadmePath =
+        getRelativePosixPath(targetPackage.readmeFile, from: example.directory);
+    final int exitCode = await processRunner.runAndStream(
+        'dart',
+        <String>[
+          'run',
+          'code_excerpt_updater',
+          '--write-in-place',
+          '--yaml',
+          '--no-escape-ng-interpolation',
+          relativeReadmePath,
+        ],
+        workingDir: example.directory);
+    return exitCode == 0;
+  }
+
+  /// Adds `code_excerpter` and `code_excerpt_updater` to [package]'s
+  /// `dev_dependencies` using path-based references to the submodule copies.
+  /// This is done on the fly rather than being checked in so that:
+  /// - Just building examples don't require everyone to check out submodules.
+  /// - Examples can be analyzed/built even on versions of Flutter that these
+  ///   submodules do not support.
+  void _addSubmoduleDependencies(RepositoryPackage package,
+      {required Directory repoRoot}) {
+    final String pubspecContents = package.pubspecFile.readAsStringSync();
+    // Save aside a copy of the current pubspec state. This allows restoration
+    // to the previous state regardless of its git status at the time the script
+    // ran.
+    package.directory
+        .childFile(_originalPubspecFilename)
+        .writeAsStringSync(pubspecContents);
+
+    // Update the actual pubspec.
```

Comment:
Have you tried using [yaml_edit](https://pub.dev/packages/yaml_edit) for these type of changes? It may make it alot easier.

Reply:
Nice tip! I think you or Aaron mentioned that package as a possibility to improve some of the existing repo tooling a while back, and then I forgot to look into it.

It's about the same amount of code, but what it's doing is much clearer and feels a lot less like a hack. We could definitely apply it elsewhere in the tool, which I'll keep in mind going forward.